*: replace `"testdata"` with testutils.TestDataPath

otan opened the pull request:
Commands:
```
ag '"testdata"' -G '_test\.go$' pkg -l | xargs sed -i '' -e 's/"testdata"/testutils.TestDataPath()/g'
ag '`testdata`' -G '_test\.go$' pkg -l | xargs sed -i '' -e 's/`testdata`,/testutils.TestDataPath(),/g'
git diff --name-only | xargs -I {} crlfmt -w {}
make bazel-generate
```

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75124)
<!-- Reviewable:end -->

otan committed:
*: replace `"testdata"` with testutils.TestDataPath

Commands:
```
ag '"testdata"' -G '_test\.go$' pkg -l | xargs sed -i '' -e 's/"testdata"/testutils.TestDataPath()/g'
ag '`testdata`' -G '_test\.go$' pkg -l | xargs sed -i '' -e 's/`testdata`,/testutils.TestDataPath(),/g'
git diff --name-only | xargs -I {} crlfmt -w {}
make bazel-generate
```

Release note: None

otan commented:
ah didn't realise this took in `t`
